### ISSUE TITLE
Ms Teams add in stuck in authorization popup for OneDrive

### ISSUE THREAD

**Issue by Floriszz** [jgraph/drawio#2394]
* [x] I agree to follow the [Code of Conduct](https://github.com/jgraph/drawio/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
* [x] I have searched the issue tracker for a feature request that matches the one I want to file, without success.

**Describe the bug**
The publisher of the Draw.io Add-In and the tab in the MsTeams channel can open the draw.io file in Ms Teams perfectly by 1 click. But Ms Team members of the channel where the draw.io file is stored can see the links in the tab in MsTeams and can click on it, but they will never get passed the authorization popup for Onedrive. The draw.io files are not in (my) Onedrive, but in the MS Teams Site - Channel which is a folder in SharePoint Site as well.

**To Reproduce**
1. Click in the Ms Teams client/PWA or browser version in the tab bar on the draw.io file
2. Get the screenshot below and never get passed it
3. See error log from the browser console

**Expected behavior**
Instead of the Authorize draw.io to access OneDrive I would expect the user to see the draw.io file in MsTeams as the authorization is already done by the publisher who added the plugin and link in the tab to the draw.io file.

**Screenshots**
screenshot added

- draw.io version:
-Created by: JGraph
Version 1.0.12

**Desktop (please complete the following information):**
 - OS: [e.g. iOS] Windows 10
 - Browser [e.g. chrome, safari] Edge and Chrome
     -Chrome - Version 95.0.4638.69 (Official Build) (64-bit)
     - Edge - Version 95.0.1020.44 (Official build) (64-bit)
 
![image](https://user-images.githubusercontent.com/19168937/140888040-a4489b51-844a-4581-9dba-047d1c290387.png)

In the console log I replaced the actual token with the following for privacy:
{TOKEN ID REMOVED FOR PRIVACY}
[Console log DrawIO Ms Teams add on authorization error tokenid replaced.txt](https://github.com/jgraph/drawio/files/7503256/Console.log.DrawIO.Ms.Teams.add.on.authorization.error.tokenid.replaced.txt)

**Comment by ashrafteleb** [comment]
@Floriszz I've tried the app again when a normal user (non-admin)

The steps when I opened a draw.io tab is as follows:

1. An error is shown that token cannot be fetched (bug)
2. Moved to another tab and back to draw.io one. Now it asked for adding draw.io app
3. Auth is opened to allow the app
4. Auth draw.io button is shown
5. Click the button and auth
6. After reload it asked for auth again showing the button. I authed again (bug)
7. Now it works normally everywhere

So, this can be a workaround (let me know if it doesn't work for you). And we'll work on errors in step 1 and 6
Unfortunately, we cannot reduce the number of authentication to only the Teams app since our website (draw.io) needs to communicate with OneDrive/SharePoint to be able to view/edit the diagrams stored there.

But, this is done only once when the app is first-time used.

**Comment by Floriszz** [comment]
Hi @ashrafteleb , sorry for my slow response. I had to find a colleague willing and having the time to test.😉.
The workaround we are able to confirm is:
1) Click on the grayed out rectangle in the screenshot above with the text 'Activate draw.io to access OneDrive'. (although grayed out, it is clickable!)
2) A popup will be shown where you can click on 'Auth'.
3) Repeat step 2 3 times(4 in total)
I can not clarify why the auth needed 4 times.

The positive side of this workaround is that when adding a new draw.io link to the tab bar, the auth is set and the user will get the draw.io file on screen right away! So the above process needs te be done only once.

**Comment by ashrafteleb** [comment]
@Floriszz Glad it works for you now, we'll investigate why the initial auth is required multiple times.

**Comment by Floriszz** [comment]
Hi @davidjgraph, today I discovered with another user, that after click on the gray rectangle with the text 'Activate draw.io to access OneDrive', a popup with white background comes up with an endless spinning circle. So not authentication is possible and the user is stuck.
This means the workaround is not working.

**Comment by davidjgraph** [comment]
Any browser extensions? Does it work private/incognito/another browser?

**Comment by adptve** [comment]
Found a work around. 
When the pop up appears in teams app. Go to a browser and enter in https://app.diagrams.net/ and from that screen authorise one drive. Return back to teams and re-add draw.io - it shall then work.
